release yaml throws lint errors

we get this lint errors in operator, this **doesn't** block anything as we fetch the yaml while building operator
but lets fix them 😄 
```
./cmd/openshift/operator/kodata/tekton-addon/pipelines-as-code/v0.14.1/release.yaml
  1:1       error    too many blank lines (1 > 0)  (empty-lines)
  1293:37   error    no new line character at the end of file  (new-line-at-end-of-file)
```

--- a/.tekton/pull-request.yaml
+++ b/.tekton/pull-request.yaml
@@ -31,9 +31,27 @@ spec:
           - name: output
             workspace: source
 
-      - name: yamllint
+      - name: generate-release-yaml
         runAfter:
           - fetchit
+        taskSpec:
+          workspaces:
+            - name: source
+          steps:
+            - name: gen-yaml
+              image: registry.access.redhat.com/ubi9/python-39
+              workingDir: $(workspaces.source.path)
+              script: |
+                set -x
+                mkdir -p bin/ # ignored in .gitignore
+                ./hack/generate-releaseyaml.sh  > bin/release.yaml
+        workspaces:
+          - name: source
+            workspace: source
+
+      - name: yamllint
+        runAfter:
+          - generate-release-yaml
         taskSpec:
           workspaces:
             - name: source
@@ -42,6 +60,7 @@ spec:
               image: quay.io/mhutter/yamllint
               workingDir: $(workspaces.source.path)
               script: |
+                set -x
                 yamllint -f parsable -c .yamllint $(find . -type f -regex ".*y[a]ml" -print)
         workspaces:
           - name: source

--- a/hack/generate-releaseyaml.sh
+++ b/hack/generate-releaseyaml.sh
@@ -36,7 +36,7 @@ fi
 
 
 for file in ${files};do
-    head -1 ${file} | grep -q -- "---" || echo -e "\n---\n"
+    sed -e '/^$/d' -e '/^#/d' ${file} | head -1 | grep -q -- "---" || echo -e "---\n"
     sed -r -e "s,(.*image:.*)ko://github.com/openshift-pipelines/pipelines-as-code/cmd/pipelines-as-code-controller.*,\1${TARGET_REPO_CONTROLLER}:${TARGET_BRANCH}\"," \
         -r -e "s,(.*image:.*)ko://github.com/openshift-pipelines/pipelines-as-code/cmd/pipelines-as-code-watcher.*,\1${TARGET_REPO_WATCHER}:${TARGET_BRANCH}\"," \
         -r -e "s,(.*image:.*)ko://github.com/openshift-pipelines/pipelines-as-code/cmd/pipelines-as-code-webhook.*,\1${TARGET_REPO_WEBHOOK}:${TARGET_BRANCH}\"," \
@@ -52,5 +52,7 @@ for file in ${files};do
         sed -ir '/^[ ]*- apiGroups:.*route.openshift.io/,/verbs.*/d' ${TMP}
     }
 
+    echo "" >> ${TMP}
+    tail -1 ${TMP} |grep -q "^$" && sed -i '$d' ${TMP} >> /tmp/aaaa
     cat ${TMP}
 done

